app/controllers/hostgroups_controller.rb

```diff
@@ -85,11 +85,11 @@ def puppetclass_parameters
   def environment_selected
     return not_found unless (@environment = Environment.find(params[:environment_id])) if params[:environment_id].to_i > 0
 
-    @hostgroup ||= Hostgroup.new
+    @hostgroup = Hostgroup.find_by_id(params[:hostgroup_id]) || Hostgroup.new
```

Comment:
@daviddavis whats the motivation for the change? note that @hostgroup can already be set before, hence you should apply your logic to that part instead of adding it here?